src/regions/PolygonRegion.js

```diff
@@ -65,225 +65,239 @@ const Model = types
-      const point = getAnchorPoint({ flattenedPoints, cursorX, cursorY });
+        removeHoverAnchor({ layer: e.currentTarget.getLayer() });
 
-      self.insertPoint(insertIdx, point[0], point[1]);
-    },
+        const { offsetX, offsetY } = e.evt;
+        const [cursorX, cursorY] = self.parent.fixZoomedCoords([offsetX, offsetY]);
+        const point = getAnchorPoint({ flattenedPoints, cursorX, cursorY });
 
-    deletePoint(point) {
-      const willNotEliminateClosedShape = self.points.length <= 3 && point.parent.closed;
-      const isLastPoint = self.points.length === 1;
-      const isSelected = self.selectedPoint === point;
-      
-      if (willNotEliminateClosedShape || isLastPoint) return;
-      if (isSelected) self.selectedPoint = null;
-      destroy(point);
-    },
+        self.insertPoint(insertIdx, point[0], point[1]);
+      },
 
-    addPoint(x, y) {
-      if (self.closed) return;
-      self._addPoint(x, y);
-    },
+      deletePoint(point) {
+        const willNotEliminateClosedShape = self.points.length <= 3 && point.parent.closed;
+        const isLastPoint = self.points.length === 1;
+        const isSelected = self.selectedPoint === point;
 
-    setPoints(points) {
-      self.points.forEach((p, idx) => {
-        p.x = points[idx * 2];
-        p.y = points[idx * 2 + 1];
-      });
-    },
+        if (willNotEliminateClosedShape || isLastPoint) return;
+        if (isSelected) self.selectedPoint = null;
+        destroy(point);
+      },
 
-    insertPoint(insertIdx, x, y) {
-      const p = {
-        id: guidGenerator(),
-        x,
-        y,
-        size: self.pointSize,
-        style: self.pointStyle,
-        index: self.points.length,
-      };
-
-      self.points.splice(insertIdx, 0, p);
-    },
+      addPoint(x, y) {
+        if (self.closed) return;
+        self._addPoint(x, y);
 
-    _addPoint(x, y) {
-      self.points.push({
-        id: guidGenerator(),
-        x,
-        y,
-        size: self.pointSize,
-        style: self.pointStyle,
-        index: self.points.length,
-      });
-    },
+        _historyPoints = [...self.points];
+      },
 
-    // @todo not used
-    // only px coordtype here
-    rotate(degree = -90) {
-      self.points.forEach(point => {
-        const p = self.rotatePoint(point, degree);
+      setPoints(points) {
+        self.points.forEach((p, idx) => {
+          p.x = points[idx * 2];
+          p.y = points[idx * 2 + 1];
+        });
+      },
 
-        point._movePoint(p.x, p.y);
-      });
-    },
+      insertPoint(insertIdx, x, y) {
+        const p = {
+          id: guidGenerator(),
+          x,
+          y,
+          size: self.pointSize,
+          style: self.pointStyle,
+          index: self.points.length,
+        };
 
-    closePoly() {
-      self.closed = true;
-    },
+        self.points.splice(insertIdx, 0, p);
+      },
 
-    canClose(x, y) {
-      if (self.points.length < 2) return false;
+      undoRedoPoints(undo = true){
```

Comment:
there are no common logic, so it would be more clear to split this method in two: undoPoint/redoPoint